### ISSUE TITLE
Falls over when processing text message with unknown origin number

### ISSUE THREAD

**Issue by einari** [IFRCGo/cbs#505]
### Description

If the origin number does not have a corresponding data collector - an exception is thrown and processing falls over. We are also trying to get the Phone Number for reporting invalid message - which kinda makes no sense if the datacollector is not even registered. The result is that we don't even report this.

```csharp
public void Process(TextMessage message)
        {
            var parsingResult = _textMessageParser.Parse(message);
            var dataCollector = _dataCollectors.GetByPhoneNumber(message.OriginNumber);  <--- Falls over
            var caseReportId = Guid.NewGuid();
            var caseReporting = _caseReportingRepository.Get(caseReportId);
            if (!parsingResult.IsValid)
            {
                ReportInvalidMessage(message, parsingResult.ErrorMessages, dataCollector, caseReporting, message.Sent);
                return;
            }
```

### Steps to recreate
1. Set a breakpoint in the Process method in TextMessageProcessor.cs in VolunteerReporting
2. Send a text message with unknown datacollector
3. See it crash

### Current behavior
Exception gets thrown and logged and swallowed.

### Expected behavior
Should report a message for unknown data collector

**Comment by Velcrow81** [comment]
Yoho. We can send text messages I assume :)

**Comment by einari** [comment]
You'd think..  :)

**Comment by Velcrow81** [comment]
Could this be related to the bug that we discovered in the event system yesterday @einari?

**Comment by einari** [comment]
It shouldn't try to get a data collector. The first it does is parse and it uses the data collector if what it parsed and it was invalid format - but what if the data collector does not exist, it can't use it for reporting invalid message. Something is suggesting we need to fine grain this a bit. Typically the case of non data collector but not valid...

**Comment by Velcrow81** [comment]
I am unable to reproduce when I use the TestDatagenerator. If I only send test text messages, they are all unknown and it works. The code handles if data collector is null by selecting the unknown and invalid report (if you drill down into the first method. But since this seems like its easy to misunderstand, it should be refactored. When you test, did you send text message through the text message controller with kafka running? I have not gotten that to work yet, so just tested with test data

**Comment by einari** [comment]
I just recalled something I changed mid-stream here and didn't test this all that much after the change. The Phone number was null coming in. That is obviously not a real scenario - so we don't need to think about that. But that would explain weirdness. 

I think that method is a bit hard to follow and it seems confusing and not clear with the hidden Null possibilities..

**Comment by Velcrow81** [comment]
I will create an issue to refactor to make it more easier to read. Also closing this bug since it is not really a bug 😊

**Comment by Velcrow81** [comment]
Im sleepy. Me good english write
